src/peergos/shared/storage/DirectS3BlockStore.java

```diff
@@ -211,7 +212,17 @@ public static Multihash keyToHash(String keyFileName) {
                             res[i] = new FragmentWithHash(new Fragment(identity.getHash()), Optional.empty());
                         }
                     return Arrays.asList(res);
+                }).thenAccept(allResults::complete)
+                .exceptionally(t -> {
+                    NetworkAccess.downloadFragments(hashes, this, monitor, spaceIncreaseFactor)
```

Comment:
Good to add a comment here to explain fallback

Reply:
Is it not clear given it is the same as the call at the beginning of the method?

Reply:
not that bothered